json_newline_writer_sink.go

```diff
@@ -0,0 +1,36 @@
+package health
+
+import "io"
+
+type JsonNewlineWriterSink struct {
+	jj *JsonWriterSink
```

Comment:
so that's way cool (and i didnt know it), but i don't think it works for the other comment about constructor method

ie: users will need to pass a a JsonWriterSink instead of io.Writer which wont work